src/Client.php

```diff
@@ -1222,8 +1222,21 @@ public function updateJob(string $name, string $image_name, array $args, array $
   /**
    * {@inheritdoc}
    */
-  public function deleteJob(string $name) {
-    return $this->apiCall(__METHOD__, $name);
+  public function deleteJob(string $name, string $label = NULL) {
```

Comment:
Hmm, the code here is completely separate. I think we're better off implementing `deleteJobByLabel(string $label)` and `getJobByLabel(string $label)`.

Reply:
Well, there are a couple more spots that use the label thing as well, so if we're changing one, lets do them all. Do that and make this MR bigger, or do them later?